src/io.jl

```diff
@@ -223,10 +221,13 @@ function prepare_gen_data(
     keys = ["coords", "volt", "tech", "cap", "pfactor", "minload", "scap",
             "wcap", "shut2loadtime", "status"]
     for p in plants
-        push!(fplants, [Dict(k => v for (k,v) in zip(keys,g)) for g in p])
-        for g in fplants[end]
-            format_gen!(g)
+        plant_dict = [Dict(k => v for (k,v) in zip(keys,g)) for g in p]
+
+        for plant in plant_dict
```

Comment:
Minor naming issue here, but a `plant_dict` represents a plant is a collection of generators, not of plants (which can either be a single or multiple generators). I'd use `for gen in plant_dict` or similar.